salt/modules/zypper.py

```diff
@@ -788,16 +789,23 @@ def mod_repo(repo, **kwargs):
 
     if kwargs.get('gpgautoimport') is True:
         global_cmd_opt.append('--gpg-auto-import-keys')
+        trigger_refresh = True
 
     if cmd_opt:
         cmd_opt = global_cmd_opt + ['mr'] + cmd_opt + [repo]
         __zypper__.refreshable.xml.call(*cmd_opt)
 
     # If repo nor added neither modified, error should be thrown
-    if not added and not cmd_opt:
+    if not added and not cmd_opt and not trigger_refresh:
```

Comment:
@isbm
I do not like the idea of dropping the specialized variable because I want to have a single source of truth.
So I want to check only once this:

```python
if kwargs.get('gpgautoimport') is True:
```
So I'm relying on that single line for adding `--gpg-auto-import-keys` option to `global_cmd_opt` and calling the refresh and not on two separate conditions